contracts/core/Dispatcher.sol

```diff
@@ -146,7 +157,20 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
         nextSequenceRecv[address(portAddress)][local.channelId] = 1;
```

Comment:
Since the tx will "never revert" now, do we want to do this only when the dapp callback succeeds?

Reply:
Ah yes, I was wondering this, but I think the reason why we want to still run the logic is that we still consider it successful transport so we still want to update state;

I want to confirm with @alfredo-stonk though

Reply:
I can see both options being valid. it's up to us to decide. Whatever we end up doing, let's document it. Maybe a comment in the code will help. Also, we need to make it explicit to devs (regardless of what we do)

Reply:
If we leave it this way, then dApp can send packets even when channel handshake fails, which is a big security loophole

Reply:
though not sure if we want to revert in the case of a callback failure in this step?

Reply:
Ok should be fixed now. 

I also did a similar thing for logic in the `acknowledgement` method to only do state updates in the case of success. Though for the `recvPacket` I left it as still updating the nextSequenceRecv mapping since I think we still consider successful transport in the case of a callback failure (and if it's gotten to this step then we know it's not an EOA)

Reply:
maybe @alfredo-stonk should verify the logic sourrounding try/catch in the `acknowledgement`, `recvPacket` and updated logic in `onConnectIbcChannel`  is indeed what we want?